packages/circuits/jwt-verifier-template.circom

```diff
@@ -332,35 +334,43 @@ template JWTVerifier(
     signal embeddedAccountCode <== Hex2FieldModular(invitationCodeLen)(invitationCodeHex);
     isCodeExist * (embeddedAccountCode - accountCode) === 0;
 
-    if (enableAnonymousDomains) {
+    if (revealEmailDomain || enableAnonymousDomains) {
         signal input emailDomainIndex;
         signal input emailDomainLength;
-        signal input anonymousDomainsTreeRoot;
-        signal input emailDomainPath[anonymousDomainsTreeHeight];
-        signal input emailDomainPathHelper[anonymousDomainsTreeHeight];
 
         var maxDomainLength = DOMAIN_MAX_BYTES();
         var maxDomainFieldLength = compute_ints_size(maxDomainLength);
 
         // Extract the domain from the email
         signal domainNameBytes[maxDomainLength] <== RevealSubstring(maxEmailLength, maxDomainLength, 0)(email, emailDomainIndex, emailDomainLength);
-        signal domainName[maxDomainFieldLength] <== Bytes2Ints(maxDomainLength)(domainNameBytes);
 
-        // Generate the leaf of the Merkle tree for the email domain
-        component domainHasher = PoseidonModular(maxDomainFieldLength);
-        for (var i = 0; i < maxDomainFieldLength; i++) {
-            domainHasher.in[i] <== domainName[i];
+        if (revealEmailDomain && !enableAnonymousDomains) {
```

Comment:
Instead of defining the condition `revealEmailDomain && !enableAnonymousDomains`, could you simply define it as`if(revealEmailDomain)`, and instead assert that `revealEmailDomain && enableAnonymousDomains` is false?
Also, how do you think about renaming `enableAnonymousDomains` to `verifyAnonymousDomains`? 
I feel that flag name does not explain the functionally clearly.